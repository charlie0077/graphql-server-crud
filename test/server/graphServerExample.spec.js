import { query } from './query'
import { mutation } from './mutation'

describe('Query Tests', query)

describe('Mutation Tests', mutation)
