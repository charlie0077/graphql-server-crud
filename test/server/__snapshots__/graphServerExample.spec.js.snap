// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation Tests test bulk delete 1`] = `
Object {
  "data": Object {
    "bulkDeletePost": Array [
      Object {
        "__typename": "Post",
        "id": "0",
        "title": "Udwirte nifenlu ubugahuf rugnu huweg raskew fawkidij lomoru.",
      },
      Object {
        "__typename": "Post",
        "id": "2222",
        "title": "new post 2222",
      },
    ],
  },
}
`;

exports[`Mutation Tests test bulk insert 1`] = `
Object {
  "data": Object {
    "bulkInsertPost": Array [
      Object {
        "__typename": "Post",
        "id": "2222",
        "title": "new post 2222",
      },
      Object {
        "__typename": "Post",
        "id": "3333",
        "title": "new post 3333",
      },
    ],
  },
}
`;

exports[`Mutation Tests test bulk update 1`] = `
Object {
  "data": Object {
    "bulkUpdatePost": Array [
      null,
      null,
    ],
  },
}
`;

exports[`Mutation Tests test delete 1`] = `
Object {
  "data": Object {
    "deletePost": Object {
      "__typename": "Post",
      "id": "1111",
      "title": "new post 1111",
    },
  },
}
`;

exports[`Mutation Tests test insert 1`] = `
Object {
  "data": Object {
    "insertPost": Object {
      "__typename": "Post",
      "id": "1111",
      "title": "new post 1111",
    },
  },
}
`;

exports[`Mutation Tests test no returning value 1`] = `
Object {
  "data": Object {
    "updatePost": null,
  },
}
`;

exports[`Mutation Tests test update 1`] = `
Object {
  "data": Object {
    "updatePost": null,
  },
}
`;

exports[`Query Tests test aggregation 1`] = `
Object {
  "data": Object {
    "queryAuthorJoinPost": Array [
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 88,
        "email": "vilurazu@vuse.tk",
        "id": "1",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 32,
        "email": "pi@ga.co.uk",
        "id": "6",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 54,
        "email": "on@puveca.in",
        "id": "8",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 35,
        "email": "cor@viiru.jp",
        "id": "9",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 33,
        "email": "ijsettav@mo.tl",
        "id": "10",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 34,
        "email": "ige@pib.ph",
        "id": "11",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 70,
        "email": "avo@ped.cc",
        "id": "12",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 61,
        "email": "de@bosdufzi.ps",
        "id": "13",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 69,
        "email": "ru@etu.er",
        "id": "14",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 89,
        "email": "kezhupur@topevgos.gov",
        "id": "15",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`Query Tests test complicate search 1`] = `
Object {
  "data": Object {
    "queryAuthor": Array [],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`Query Tests test limit and offset 1`] = `
Object {
  "data": Object {
    "queryPost": Array [
      Object {
        "__typename": "PostWithAggregation",
        "id": "0",
        "title": "Udwirte nifenlu ubugahuf rugnu huweg raskew fawkidij lomoru.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "1",
        "title": "Igijuto ne ruj mab ipju dan ros benozezu.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "2",
        "title": "Pojawri lotnemip ipemuipi ridvuh oddale judiniin ve cerhub.",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`Query Tests test multiple queries in a request 1`] = `
Object {
  "data": Object {
    "getAuthor": Object {
      "__typename": "Author",
      "email": "vilurazu@vuse.tk",
      "id": "1",
    },
    "queryPost": Array [
      Object {
        "__typename": "PostWithAggregation",
        "id": "99",
        "title": "Nokcar conuk mi rub rokdul en tunnegov zojan.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "98",
        "title": "Jeaka govut iva liczizig denbu ecjob fov amofi.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "97",
        "title": "Buufa judosbup ec mepecmu nezojeave sehgieru foakepi emgotez.",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`Query Tests test nested query 1`] = `
Object {
  "data": Object {
    "queryPost": Array [
      Object {
        "__typename": "PostWithAggregation",
        "author": Object {
          "__typename": "Author",
          "company": Object {
            "__typename": "Company",
            "domain": "nouco.cn",
          },
          "email": "ur@moej.wf",
        },
        "id": "2",
        "title": "Pojawri lotnemip ipemuipi ridvuh oddale judiniin ve cerhub.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "author": Object {
          "__typename": "Author",
          "company": Object {
            "__typename": "Company",
            "domain": "nitdopfiz.ro",
          },
          "email": "vilurazu@vuse.tk",
        },
        "id": "3",
        "title": "Gokgor kezli lazgoh gaw nite mewseun do supma.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "author": Object {
          "__typename": "Author",
          "company": Object {
            "__typename": "Company",
            "domain": "nouco.cn",
          },
          "email": "ur@moej.wf",
        },
        "id": "4",
        "title": "Totunusa kazaha ruguk di zuvon mojfogow datiha ewa.",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`Query Tests test simple query 1`] = `
Object {
  "data": Object {
    "getAuthor": Object {
      "__typename": "Author",
      "email": "vilurazu@vuse.tk",
      "id": "1",
    },
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`Query Tests test simple query 1 1`] = `
Object {
  "data": Object {
    "getUserInfo": Object {
      "__typename": "UserInfo",
      "id": "1",
    },
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`Query Tests test sort 1`] = `
Object {
  "data": Object {
    "queryPost": Array [
      Object {
        "__typename": "PostWithAggregation",
        "id": "99",
        "title": "Nokcar conuk mi rub rokdul en tunnegov zojan.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "98",
        "title": "Jeaka govut iva liczizig denbu ecjob fov amofi.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "97",
        "title": "Buufa judosbup ec mepecmu nezojeave sehgieru foakepi emgotez.",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;
